src/tree_types.ml

```diff
@@ -0,0 +1,155 @@
+(** Miscellaneous types and utilities for decision trees. *)
+open Extra
+
+(** {3 Tree constructors} *)
+
+(** Define a simpler representation of terms. *)
+module Treecons =
+struct
+  (** A constructor is the representation of a symbol along with the number of
+      arguments to which it is applied. *)
+  type t =
+    | Symb of (int * string * Files.module_path)
```

Comment:
Remove the parentheses here, otherwise there is an extra indirection in the memory representation (i.e., it is a constructor with one tuple as argument instead of being a constructor with three arguments).